frontend/src/employee-mobile-frontend/components/ensureAuthenticated.tsx

```diff
@@ -4,16 +4,23 @@
 
 import React, { ComponentType, useContext } from 'react'
 import { Redirect } from 'react-router-dom'
+import { FullScreenDimmedSpinner } from 'lib-components/molecules/FullScreenDimmedSpinner'
 import { UserContext } from '../state/user'
 
 export default function ensureAuthenticated<P>(Component: ComponentType<P>) {
   return function Authenticated(props: P) {
-    const { loggedIn } = useContext(UserContext)
+    const { loggedIn, user } = useContext(UserContext)
 
     if (!loggedIn) {
       return <Redirect to="" />
     }
 
-    return <Component {...props} />
+    const showSpinner = user.isLoading || (user.isSuccess && user.isReloading)
```

Comment:
Are you sure you want to show the full screen spinner when reloading the user? 🤔

Reply:
Well, tbh we can get rid of it 😄

Reply:
Come to think of it, the loading state and spinner is already handled in the landing page component. We can just remove the spinner from here altogether.